samza-azure/src/main/java/org/apache/samza/system/eventhub/SamzaEventHubClientManager.java

```diff
@@ -52,29 +54,36 @@
   private final String sasKey;
   private final RetryPolicy retryPolicy;
 
-  public SamzaEventHubClientManager(String eventHubNamespace, String entityPath, String sasKeyName, String sasKey) {
+  public SamzaEventHubClientManager(String eventHubNamespace, String entityPath, String sasKeyName, String sasKey,
+      Integer numClientThreads) {
     this(eventHubNamespace, entityPath, sasKeyName, sasKey,
-            new RetryExponential(MIN_RETRY_BACKOFF, MAX_RETRY_BACKOFF, MAX_RETRY_COUNT, SAMZA_EVENTHUB_RETRY));
+            new RetryExponential(MIN_RETRY_BACKOFF, MAX_RETRY_BACKOFF, MAX_RETRY_COUNT, SAMZA_EVENTHUB_RETRY), numClientThreads);
   }
 
   public SamzaEventHubClientManager(String eventHubNamespace, String entityPath, String sasKeyName, String sasKey,
-                                    RetryPolicy retryPolicy) {
+                                    RetryPolicy retryPolicy, int numClientThreads) {
     this.eventHubNamespace = eventHubNamespace;
     this.entityPath = entityPath;
     this.sasKeyName = sasKeyName;
     this.sasKey = sasKey;
     this.retryPolicy = retryPolicy;
+    this.numClientThreads = numClientThreads;
   }
 
   @Override
   public void init() {
     String remoteHost = String.format(EVENTHUB_REMOTE_HOST_FORMAT, eventHubNamespace);
     try {
       ConnectionStringBuilder connectionStringBuilder =
-              new ConnectionStringBuilder(eventHubNamespace, entityPath, sasKeyName, sasKey);
+              new ConnectionStringBuilder()
+          .setNamespaceName(eventHubNamespace)
+          .setEventHubName(entityPath)
+          .setSasKeyName(sasKeyName)
+          .setSasKey(sasKey);
 
-      eventHubClient = EventHubClient.createFromConnectionStringSync(connectionStringBuilder.toString(), retryPolicy);
-    } catch (IOException | ServiceBusException e) {
+      eventHubClient = EventHubClient.createSync(connectionStringBuilder.toString(), retryPolicy,
+          Executors.newFixedThreadPool(numClientThreads));
```

Comment:
Does this need to be shut down when this client is stopped? Do the threads need to be daemon? Should also name the threads in this pool. Convention: "Samza EventHubClient Thread-%d"